flowredux/src/commonMain/kotlin/com/freeletics/flowredux/dsl/State.kt

```diff
@@ -26,25 +27,17 @@ public class State<InputState : Any>(
      * Use this function if you want to override the previous state with another state based on
      * the current state.
      *
-     * [snapshot] should never be accessed within [reducer].
+     * [snapshot] should never be accessed within [reducer]. Use the `this` of the lambda which
+     * is guaranteed to be the current state at the time of execution instead.
      */
     public fun <S : Any> override(reducer: InputState.() -> S): ChangeState<S> {
         return UnsafeMutateState(reducer)
     }
 
     /**
-     * Sets a new state by directly override any previous state. If the new state requires values
-     * from the current state prefer using the `override` function that has a `reducer` parameter.
-     */
-    public fun <S : Any> override(newState: S): ChangeState<S> {
```

Comment:
I know that we spoke about removing or replacing this with `.override { ...}` but just for the records (and because I have a terrible memory): what do we want to achieve here with removing this?

Reply:
I'm fine with removing it, I just cant remember how our discussion about this went 😄

Reply:
Also not sure about the discussion, the general motivation is what I wrote in the description
> It also removes state.override(newState) to force using the lambda version state.override { ... }. The former leads to the new state often being created by reading the snapshot which can lead to bugs because the snapshot can be outdated. The version with lambda gives you the current state as this and makes this less likely from what I could observe. Also makes it easier to implement the lint checks because we can enforce that new state is created inside the lambda and that the snapshot isn't accessed in there.

During the migration to `State` I saw others automatically replacing something like
```
val newState = MyState(stateSnapshot.something, ...)
return MutateState { newState }
```
with the correct approach of building the state inside `mutate` or `override`. The simple `override` still invites to doing the wrong approach above.

Reply:
Ah yea, thanks for the reminder 🙏